src/Sulu/Bundle/SnippetBundle/Content/SnippetDataProvider.php

```diff
@@ -220,6 +240,37 @@ function(
         );
     }
 
+    /**
+     * @return array<int, array<string, string>>
+     */
+    private function getTypes(): array
+    {
+        $types = [];
+        if ($this->tokenStorage && null !== $this->tokenStorage->getToken()) {
+            // the user is required for the metadata locale because of the translated template titles
+            $user = $this->tokenStorage->getToken()->getUser();
+
+            // user is "anon." if the user is not logged in
+            // this happens on the initial admin request (/admin) if the AudienceTargeting is deactivated
+            if (!$user instanceof UserInterface) {
+                return $types;
+            }
+
+            try {
+                /** @var TypedFormMetadata $metadata */
+                $metadata = $this->formMetadataProvider->getMetadata('snippet', $user->getLocale(), []);
```

Comment:
We have to handle the case in which the `formMetadataProvider` is `null`, right?